### PR TITLE
Patch/pooling

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

File: src/main/groovy/life/qbic/utils/NanoporeParser.groovy [hunk 1]
```diff
@@ -34,17 +34,16 @@ class NanoporeParser {
...
-
             // Step5: Create the final OxfordNanoporeExperiment from the map
             OxfordNanoporeExperiment convertedExperiment = OxfordNanoporeExperiment.create(finalMap)
             return convertedExperiment
         } catch (ValidationException validationException) {
             log.error("Specified directory could not be validated")
             // we have to fetch all validation exceptions
-            def causes = validationException.getCausingExceptions().collect{ it.message }.join("\n")
+            def causes = validationException.getAllMessages().collect{ it }.join("\n")
+//            def causes = validationException.getCausingExceptions().collect{ it.message }.join("\n")
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/utils/NanoporeParser.groovy:46]: Why this change here? 
```suggestion
            def causes = validationException.getAllMessages().collect{ it }.join("\n")
```
```suggestion
            def causes = validationException.getCausingExceptions().collect{ it.message }.join("\n")
```

File: src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy [hunk 2]
```diff
@@ -10,62 +10,82 @@ import java.text.ParseException
...
+    
+    assert experiment instanceof OxfordNanoporeExperiment
+    // Check that the metadata from the report file has been retrieved
+    assert measurement.getMachineHost() == "PCT0094"
+    // Check that the metadata from the summary file has been retrieved
+    assert measurement.getLibraryPreparationKit() == "SQK-LSK109"
+    // check that multiple pooled samples are contained
+    assert keys.size() > 1
+    // ...and that they are linked to two different folder structures
+    assert measurement.getRawDataPerSample(experiment).get(keys.get(0)) != measurement.getRawDataPerSample(experiment).get(keys.get(1))
```
Comment (KochTobi) [inline src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy:45]:
```suggestion
    assert rawDataPerSample.size() > 1
    // ...and that they are linked to two different folder structures
    assert rawDataPerSample.get(0) != rawDataPerSample.get(1)
```

File: src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy [hunk 3]
```diff
@@ -10,62 +10,82 @@ import java.text.ParseException
...
-        then:
-        thrown(ValidationException)
-    }
+  def "parsing a pooled file structure returns an OxfordNanoporeExperiment Object"() {
+    given:
+    def pathToDirectory = Paths.get(exampleDirectoriesRoot, "validates/QABCD001AB_E12A345a01_PAE12345_pooled")
+    when:
+    def experiment = NanoporeParser.parseFileStructure(pathToDirectory)
+    def measurement = experiment.getMeasurements().get(0)
+    def List<String> keys = new ArrayList<>(measurement.getRawDataPerSample(experiment).keySet())
```
Comment (KochTobi) [inline src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy:34]:
```suggestion
   def rawDataPerSample = measurement.getRawDataPerSample(experiment).keySet()
```

File: src/main/groovy/life/qbic/utils/NanoporeParser.groovy (hunk 1 above)
Comment (wow-such-code) [reply src/main/groovy/life/qbic/utils/NanoporeParser.groovy:46]: getCausingExceptions is not as verbose as it sounds

Comment (KochTobi) [reply src/main/groovy/life/qbic/utils/NanoporeParser.groovy:46]: Ok then just remove the old stuff.

File: src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy (hunk 2 above)
Comment (wow-such-code) [reply src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy:45]: rawDataPerSample is a map of sample codes, i can't test the folders this way

Comment (wow-such-code) [reply src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy:45]: a list*

Comment (KochTobi) [reply src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy:45]: Try using `.entrySet()` instead of `keySet()`

Comment (KochTobi) [reply src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy]: better even use the correct and expected keys ;)
